app/controllers/admins_controller.rb

```diff
@@ -183,6 +168,140 @@ def default_recording_visibility
                                                I18n.t("administrator.site_settings.recording_visibility.warning") }
   end
 
+  # ROLES
+
+  # GET /admins/roles
+  def roles
+    @roles = Role.editable_roles(@user_domain)
+
+    if @roles.count.zero?
+      Role.create_default_roles(@user_domain)
+      @roles = Role.editable_roles(@user_domain)
+    end
+
+    @selected_role = if params[:selected_role].nil?
+                        @roles.find_by(name: 'user')
+                      else
+                        @roles.find(params[:selected_role])
+                     end
+  end
+
+  # POST /admin/role
+  def new_role
```

Comment:
Can you add some comments to this function? It'll help explain what is going on exactly